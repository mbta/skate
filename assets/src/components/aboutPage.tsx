import React, { ReactElement } from "react"

const AboutPage = (): ReactElement<HTMLDivElement> => (
  <div className="c-page c-page--about">
    <div className="c-page__container">
      <h1 className="c-page__title">About Skate</h1>
      <p>
        Skate is an active work-in-progress. In the coming months, you’ll notice
        new tools like shuttle tracking. You can also expect changes and
        refinements to the route ladders as we continue building.
      </p>
      <div className="c-page__section">
        <h2 className="c-page__header">Device Support</h2>
        <p>Need tablet support? Call the MBTA Help Desk at x5761.</p>
      </div>
      <div className="c-page__section">
        <h2 className="c-page__header">Frequently Asked Questions</h2>
        <p>
          Have questions about using Skate? We’re making updates all the time.
          Read the latest news and refresh your training by visiting our FAQ.
        </p>
        <p>
          <a
            className="c-page__link"
            href="https://docs.google.com/document/d/e/2PACX-1vQVzXUcQWg-nKCkb8CyModNUCXDHHRxgqPdtlkm1GQOK9aCrGMSJAoC2J8JuuGFPGnPkcetrZEk6ERH/pub"
            target="_blank"
          >
            FAQ
          </a>
        </p>
      </div>
      <div className="c-page__section">
        <h2 className="c-page__header">Feedback</h2>
        <p>
          Have feedback, a suggestion, or need other support related to Skate?
          Use our feedback form and someone on our team will be in touch with
          you.
        </p>
        <p>
          <a
            className="c-page__link"
            href="https://form.asana.com?hash=e0ca8ee9a5fbde7691ba0ff428c826eba5a9700b99249e26124dd33d13b511ca&id=1155343357379913"
            target="_blank"
          >
            Give Feedback
          </a>
        </p>
      </div>
    </div>
  </div>
)

export default AboutPage
