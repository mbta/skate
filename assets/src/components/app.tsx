import React, { ReactElement, useContext } from "react"
import { BrowserRouter, Route as BrowserRoute } from "react-router-dom"
import { SocketContext } from "../contexts/socketContext"
import { StateDispatchContext } from "../contexts/stateDispatchContext"
import useAppcues from "../hooks/useAppcues"
import { ConnectionStatus } from "../hooks/useSocket"
import DataStatusBanner from "./dataStatusBanner"
import DisconnectedModal from "./disconnectedModal"
import LadderPage from "./ladderPage"
import SearchPage from "./searchPage"
import SettingsPage from "./settingsPage"
import ShuttleMapPage from "./shuttleMapPage"
import TabBar from "./tabBar"
import { Notifications } from "./notifications"

const AppRoutes = () => {
  useAppcues()

  const [{ pickerContainerIsVisible }] = useContext(StateDispatchContext)
  const { connectionStatus } = useContext(SocketContext)

  return (
    <div className="m-app">
      <div className="m-app__banner">
        <DataStatusBanner />
      </div>
      <div className="m-app__main">
        <Notifications />
        <TabBar pickerContainerIsVisible={pickerContainerIsVisible} />
        <BrowserRoute exact={true} path="/" component={LadderPage} />
        <BrowserRoute
          exact={true}
          path="/shuttle-map"
          component={ShuttleMapPage}
        />
        <BrowserRoute exact={true} path="/settings" component={SettingsPage} />
        <BrowserRoute exact={true} path="/search" component={SearchPage} />
        {connectionStatus === ConnectionStatus.Disconnected ? (
          <DisconnectedModal />
        ) : null}
      </div>
    </div>
  )
}

const App = (): ReactElement<HTMLDivElement> => {
  return (
    <BrowserRouter>
      <AppRoutes />
    </BrowserRouter>
  )
}

export default App
