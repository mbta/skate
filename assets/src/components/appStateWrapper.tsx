import React from "react"
import { StateDispatchProvider } from "../contexts/stateDispatchContext"
import { TripsByIdProvider } from "../contexts/tripsByIdContext"
import { VehiclesByRouteIdProvider } from "../contexts/vehiclesByRouteIdContext"
import usePersistedStateReducer from "../hooks/usePersistedStateReducer"
import useSocket from "../hooks/useSocket"
import useTrips from "../hooks/useTrips"
import useVehicles from "../hooks/useVehicles"
import { TripsById } from "../schedule.d"
import { initialState, reducer } from "../state"
import App from "./app"

const AppStateWrapper = (): JSX.Element => {
  const [state, dispatch] = usePersistedStateReducer(reducer, initialState)
  const { selectedRouteIds } = state

  const tripsById: TripsById = useTrips(selectedRouteIds)
  const socket = useSocket()
  const vehiclesByRouteId = useVehicles(socket, selectedRouteIds)

  return (
    <StateDispatchProvider state={state} dispatch={dispatch}>
      <TripsByIdProvider tripsById={tripsById}>
        <VehiclesByRouteIdProvider vehiclesByRouteId={vehiclesByRouteId}>
          <App />
        </VehiclesByRouteIdProvider>
      </TripsByIdProvider>
    </StateDispatchProvider>
  )
}

export default AppStateWrapper
