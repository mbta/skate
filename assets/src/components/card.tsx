import React from "react"
import { formattedTimeDiffUnderThreshold } from "../util/dateTime"
import { unreadIcon } from "../helpers/icon"
import CloseButton from "./closeButton"

export type CardStyle = "kiwi" | "white"

interface CardProps {
  style: CardStyle
  currentTime?: Date
  openCallback?: () => void
  closeCallback?: () => void
  isUnread?: boolean
  additionalClass?: string
  title: string
  time?: Date
  noFocusOrHover?: boolean
}

export const Card: React.FC<CardProps> = ({
  children,
  style,
  currentTime,
  openCallback,
  closeCallback,
  isUnread,
  additionalClass,
  title,
  time,
  noFocusOrHover,
}) => {
  const innerLeftContent = (
    <>
      <div className="m-card__top-row">
        <div className="m-card__title">
          {isUnread ? unreadIcon() : null}
          {title}
        </div>
        {currentTime && time ? (
          <div className="m-card__time">
            {formattedTimeDiffUnderThreshold(currentTime, time, 60)}
          </div>
        ) : null}
      </div>
      <div className="m-card__contents">{children}</div>
    </>
  )

  return (
    <div
      className={
        `m-card m-card--${style}` +
        (additionalClass ? " " + additionalClass : "") +
        (noFocusOrHover ? " m-card--no-focus-or-hover" : "") +
        (!isUnread ? " m-card--read" : "")
      }
    >
      {openCallback ? (
        <button className="m-card__left" onClick={openCallback}>
          {innerLeftContent}
        </button>
      ) : (
        <div className="m-card__left">{innerLeftContent}</div>
      )}
      {closeCallback ? (
        <div className="m-card__right">
          <CloseButton closeButtonType="xl_green" onClick={closeCallback} />
        </div>
      ) : null}
    </div>
  )
}

export const CardBody: React.FC = ({ children }) => (
  <div className="m-card__body">{children}</div>
)

export interface Property {
  label: string
  value: string | null
  sensitive?: boolean
}

export interface CardPropertiesProps {
  properties: Property[]
}

export const CardProperties: React.VFC<CardPropertiesProps> = ({
  properties,
}: CardPropertiesProps) => {
  return (
    <ul className="m-card__properties">
      {properties.map((property) =>
        property.value ? (
          <li key={property.label}>
            <span className="m-card__properties-label">{property.label}</span>
            <span
              className={
                "m-card__properties-value" +
                (property.sensitive
                  ? " m-card__properties-value--sensitive"
                  : "")
              }
            >
              {property.value}
            </span>
          </li>
        ) : null
      )}
    </ul>
  )
}
