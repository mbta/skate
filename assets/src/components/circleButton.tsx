import React, { ReactNode } from "react"
import { joinClasses } from "../helpers/dom"

export const CircleButton = ({
  children,
  onClick,
  isActive,
  title,
}: {
  children?: ReactNode
  onClick?: () => void
  isActive: boolean
  title?: string
}) => (
  <button
    className={joinClasses([
      "c-circle-button",
      isActive && "c-circle-button--active",
    ])}
    onClick={onClick}
    title={title}
    aria-pressed={isActive}
  >
    <div className="c-circle-button__content">{children}</div>
  </button>
)
