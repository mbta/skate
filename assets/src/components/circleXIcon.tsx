import React, { ComponentPropsWithoutRef } from "react"

/**
 * Interactive variant of the `CircleXIcon`.
 *
 * Implements focus and hover states via CSS and SVG.
 */
export const CircleXIcon = (props: ComponentPropsWithoutRef<"svg">) => (
  <svg
    className="c-circle-x-icon"
    viewBox="0 0 48 48"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <circle
      cx="50%"
      cy="50%"
      r="75%"
      className="c-circle-x-icon__focus-circle"
    />
    <circle cx="50%" cy="50%" r="75%" className="c-circle-x-icon__white-fill" />
    <path
      className="c-circle-x-icon__path"
      d="m24 .05a24 24 0 1 0 24 23.95 24 24 0 0 0 -24-23.95zm13.19 32.54a3.33 3.33 0 1 1 -4.73 4.69l-8.46-8.57-8.57 8.48a3.33 3.33 0 1 1 -4.69-4.73l8.55-8.46-8.48-8.57a3.33 3.33 0 1 1 4.73-4.69l8.46 8.55 8.57-8.48a3.33 3.33 0 0 1 2.35-1 3.35 3.35 0 0 1 3.33 3.35 3.33 3.33 0 0 1 -1 2.35l-8.54 8.49z"
    />
  </svg>
)
