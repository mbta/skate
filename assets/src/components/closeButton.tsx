import React from "react"
import { closeIcon } from "../helpers/icon"

interface Props {
  onClick: () => void
}

const CloseButton = ({ onClick }: Props) => (
  <button className="m-close-button" onClick={onClick}>
    {closeIcon("m-close-button__icon")}
  </button>
)

export default CloseButton
