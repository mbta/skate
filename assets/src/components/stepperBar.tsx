import useComponentSize from "@rehooks/component-size"
import React, { useRef } from "react"
import { SvgProps } from "../helpers/bsIcons"

export const StepperBar = ({
  totalSteps,
  currentStep,
}: {
  totalSteps: number
  currentStep: number
}) => {
  const stepperRef = useRef(null)
  const { width } = useComponentSize(stepperRef)

  const stepWidth = width / totalSteps

  const stepper =
    currentStep == totalSteps ? (
      <FilledStepper stepWidth={stepWidth} stepCount={currentStep} />
    ) : (
      <>
        <FilledStepper
          stepWidth={stepWidth}
          stepCount={currentStep}
          isPartial={true}
        />
        <UnfilledStepper
          stepWidth={stepWidth}
          stepCount={totalSteps - currentStep}
        />
      </>
    )

  return (
    <div className="c-stepper-bar d-grid" ref={stepperRef}>
      <StepperBusIcon style={{ marginLeft: stepWidth * currentStep - 66 }} />
      <div className="d-flex">{stepper}</div>
    </div>
  )
}

const barHeight = 8
const grayFill = "#D4D7DB"
const purpleFill = "#6D39AC"

const FilledStepper = ({
  stepWidth: stepWidth,
  stepCount: stepCount,
  isPartial: isPartial,
}: {
  stepWidth: number
  stepCount: number
  isPartial?: boolean
}) => (
  <svg width={stepWidth * stepCount} height={barHeight}>
    {isPartial && (
      <rect
        width={stepWidth * stepCount}
        height={barHeight}
        x={barHeight}
        fill={grayFill}
      />
    )}
    <rect
      width={stepWidth * stepCount}
      height={barHeight}
      rx={barHeight / 2}
      ry={barHeight / 2}
      fill={purpleFill}
    />
    <StopDots stepWidth={stepWidth} numDots={stepCount} />
  </svg>
)

const UnfilledStepper = ({
  stepWidth: stepWidth,
  stepCount: stepCount,
}: {
  stepWidth: number
  stepCount: number
}) => (
  <svg width={stepWidth * stepCount} height={barHeight}>
    <rect width={barHeight} height={barHeight} fill={grayFill} />
    <rect
      width={stepWidth * stepCount}
      height={barHeight}
      rx={barHeight / 2}
      ry={barHeight / 2}
      fill={grayFill}
    />
    <StopDots stepWidth={stepWidth} numDots={stepCount} />
  </svg>
)

const StopDots = ({
  stepWidth: stepWidth,
  numDots: numDots,
}: {
  stepWidth: number
  numDots: number
}) => {
  const dots = []
  for (let i = 1; i < numDots + 1; i++) {
    dots.push(
      <ellipse
        key={`stop-dot-${i}`}
        rx={barHeight / 4}
        ry={barHeight / 4}
        cx={stepWidth * i - barHeight * 0.75}
        cy={barHeight / 2}
        fill="white"
      />
    )
  }

  return <>{dots}</>
}

const StepperBusIcon = (props: SvgProps) => (
  <svg
    width="66"
    height="20"
    viewBox="0 0 66 20"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      d="M64.5 16.0624H8.5L1 14.5625C1 5.3625 1.33333 2.72902 1.5 2.56228H3C3.27614 2.56228 3.4973 2.33744 3.51288 2.06174C3.57569 0.949877 3.85283 0.915105 4 1.06228H56L58.3212 1.99089C58.4393 2.03815 58.5654 2.06244 58.6926 2.06244H61C64.6222 2.06244 64.8333 11.5624 64.5 16.0624Z"
      fill="white"
      stroke="black"
      strokeWidth="2"
    />
    <circle
      cx="17"
      cy="15.5625"
      r="2.5"
      fill="white"
      stroke="black"
      strokeWidth="2"
    />
    <circle
      cx="55"
      cy="15.5625"
      r="2.5"
      fill="white"
      stroke="black"
      strokeWidth="2"
    />
    <mask id="path-4-inside-1_3975_16368" fill="white">
      <path d="M8 5.9375C8 5.38522 8.44772 4.9375 9 4.9375H22C22.5523 4.9375 23 5.38522 23 5.9375V8.9375C23 9.48978 22.5523 9.9375 22 9.9375H12C9.79086 9.9375 8 8.14664 8 5.9375Z" />
    </mask>
    <path
      d="M8 5.9375C8 5.38522 8.44772 4.9375 9 4.9375H22C22.5523 4.9375 23 5.38522 23 5.9375V8.9375C23 9.48978 22.5523 9.9375 22 9.9375H12C9.79086 9.9375 8 8.14664 8 5.9375Z"
      fill="white"
      stroke="black"
      strokeWidth="4"
      mask="url(#path-4-inside-1_3975_16368)"
    />
    <mask id="path-5-inside-2_3975_16368" fill="white">
      <rect x="25" y="4.9375" width="30" height="6" rx="1" />
    </mask>
    <rect
      x="25"
      y="4.9375"
      width="30"
      height="6"
      rx="1"
      fill="white"
      stroke="black"
      strokeWidth="4"
      mask="url(#path-5-inside-2_3975_16368)"
    />
    <mask id="path-6-inside-3_3975_16368" fill="white">
      <path d="M56.5 6.0625C56.5 5.51022 56.9477 5.0625 57.5 5.0625H59.5C60.6046 5.0625 61.5 5.95793 61.5 7.0625V9.0625C61.5 9.61478 61.0523 10.0625 60.5 10.0625H57.5C56.9477 10.0625 56.5 9.61478 56.5 9.0625V6.0625Z" />
    </mask>
    <path
      d="M56.5 6.0625C56.5 5.51022 56.9477 5.0625 57.5 5.0625H59.5C60.6046 5.0625 61.5 5.95793 61.5 7.0625V9.0625C61.5 9.61478 61.0523 10.0625 60.5 10.0625H57.5C56.9477 10.0625 56.5 9.61478 56.5 9.0625V6.0625Z"
      fill="white"
      stroke="black"
      strokeWidth="4"
      mask="url(#path-6-inside-3_3975_16368)"
    />
  </svg>
)
