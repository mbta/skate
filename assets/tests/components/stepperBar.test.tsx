import { describe, test, expect } from "@jest/globals"
import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/jest-globals"
import { StepperBar } from "../../src/components/stepperBar"

describe("StepperBar", () => {
  test("renders 1/3rd of the way", () => {
    const { baseElement } = render(
      <StepperBar totalSteps={3} currentStep={1} />
    )

    expect(baseElement).toMatchSnapshot()
  })

  test("renders when stepper bar is complete", () => {
    const { baseElement } = render(
      <StepperBar totalSteps={3} currentStep={1} />
    )

    expect(baseElement).toMatchSnapshot()
  })
})
