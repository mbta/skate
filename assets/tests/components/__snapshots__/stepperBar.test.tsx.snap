// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepperBar renders 1/3rd of the way 1`] = `
<body>
  <div>
    <div
      class="c-stepper-bar d-grid"
    >
      <svg
        fill="none"
        height="20"
        style="margin-left: -66px;"
        viewBox="0 0 66 20"
        width="66"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M64.5 16.0624H8.5L1 14.5625C1 5.3625 1.33333 2.72902 1.5 2.56228H3C3.27614 2.56228 3.4973 2.33744 3.51288 2.06174C3.57569 0.949877 3.85283 0.915105 4 1.06228H56L58.3212 1.99089C58.4393 2.03815 58.5654 2.06244 58.6926 2.06244H61C64.6222 2.06244 64.8333 11.5624 64.5 16.0624Z"
          fill="white"
          stroke="black"
          stroke-width="2"
        />
        <circle
          cx="17"
          cy="15.5625"
          fill="white"
          r="2.5"
          stroke="black"
          stroke-width="2"
        />
        <circle
          cx="55"
          cy="15.5625"
          fill="white"
          r="2.5"
          stroke="black"
          stroke-width="2"
        />
        <mask
          fill="white"
          id="path-4-inside-1_3975_16368"
        >
          <path
            d="M8 5.9375C8 5.38522 8.44772 4.9375 9 4.9375H22C22.5523 4.9375 23 5.38522 23 5.9375V8.9375C23 9.48978 22.5523 9.9375 22 9.9375H12C9.79086 9.9375 8 8.14664 8 5.9375Z"
          />
        </mask>
        <path
          d="M8 5.9375C8 5.38522 8.44772 4.9375 9 4.9375H22C22.5523 4.9375 23 5.38522 23 5.9375V8.9375C23 9.48978 22.5523 9.9375 22 9.9375H12C9.79086 9.9375 8 8.14664 8 5.9375Z"
          fill="white"
          mask="url(#path-4-inside-1_3975_16368)"
          stroke="black"
          stroke-width="4"
        />
        <mask
          fill="white"
          id="path-5-inside-2_3975_16368"
        >
          <rect
            height="6"
            rx="1"
            width="30"
            x="25"
            y="4.9375"
          />
        </mask>
        <rect
          fill="white"
          height="6"
          mask="url(#path-5-inside-2_3975_16368)"
          rx="1"
          stroke="black"
          stroke-width="4"
          width="30"
          x="25"
          y="4.9375"
        />
        <mask
          fill="white"
          id="path-6-inside-3_3975_16368"
        >
          <path
            d="M56.5 6.0625C56.5 5.51022 56.9477 5.0625 57.5 5.0625H59.5C60.6046 5.0625 61.5 5.95793 61.5 7.0625V9.0625C61.5 9.61478 61.0523 10.0625 60.5 10.0625H57.5C56.9477 10.0625 56.5 9.61478 56.5 9.0625V6.0625Z"
          />
        </mask>
        <path
          d="M56.5 6.0625C56.5 5.51022 56.9477 5.0625 57.5 5.0625H59.5C60.6046 5.0625 61.5 5.95793 61.5 7.0625V9.0625C61.5 9.61478 61.0523 10.0625 60.5 10.0625H57.5C56.9477 10.0625 56.5 9.61478 56.5 9.0625V6.0625Z"
          fill="white"
          mask="url(#path-6-inside-3_3975_16368)"
          stroke="black"
          stroke-width="4"
        />
      </svg>
      <div
        class="d-flex"
      >
        <svg
          height="8"
          width="0"
        >
          <rect
            fill="#D4D7DB"
            height="8"
            width="0"
            x="8"
          />
          <rect
            fill="#6D39AC"
            height="8"
            rx="4"
            ry="4"
            width="0"
          />
          <ellipse
            cx="-6"
            cy="4"
            fill="white"
            rx="2"
            ry="2"
          />
        </svg>
        <svg
          height="8"
          width="0"
        >
          <rect
            fill="#D4D7DB"
            height="8"
            width="8"
          />
          <rect
            fill="#D4D7DB"
            height="8"
            rx="4"
            ry="4"
            width="0"
          />
          <ellipse
            cx="-6"
            cy="4"
            fill="white"
            rx="2"
            ry="2"
          />
          <ellipse
            cx="-6"
            cy="4"
            fill="white"
            rx="2"
            ry="2"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`StepperBar renders when stepper bar is complete 1`] = `
<body>
  <div>
    <div
      class="c-stepper-bar d-grid"
    >
      <svg
        fill="none"
        height="20"
        style="margin-left: -66px;"
        viewBox="0 0 66 20"
        width="66"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M64.5 16.0624H8.5L1 14.5625C1 5.3625 1.33333 2.72902 1.5 2.56228H3C3.27614 2.56228 3.4973 2.33744 3.51288 2.06174C3.57569 0.949877 3.85283 0.915105 4 1.06228H56L58.3212 1.99089C58.4393 2.03815 58.5654 2.06244 58.6926 2.06244H61C64.6222 2.06244 64.8333 11.5624 64.5 16.0624Z"
          fill="white"
          stroke="black"
          stroke-width="2"
        />
        <circle
          cx="17"
          cy="15.5625"
          fill="white"
          r="2.5"
          stroke="black"
          stroke-width="2"
        />
        <circle
          cx="55"
          cy="15.5625"
          fill="white"
          r="2.5"
          stroke="black"
          stroke-width="2"
        />
        <mask
          fill="white"
          id="path-4-inside-1_3975_16368"
        >
          <path
            d="M8 5.9375C8 5.38522 8.44772 4.9375 9 4.9375H22C22.5523 4.9375 23 5.38522 23 5.9375V8.9375C23 9.48978 22.5523 9.9375 22 9.9375H12C9.79086 9.9375 8 8.14664 8 5.9375Z"
          />
        </mask>
        <path
          d="M8 5.9375C8 5.38522 8.44772 4.9375 9 4.9375H22C22.5523 4.9375 23 5.38522 23 5.9375V8.9375C23 9.48978 22.5523 9.9375 22 9.9375H12C9.79086 9.9375 8 8.14664 8 5.9375Z"
          fill="white"
          mask="url(#path-4-inside-1_3975_16368)"
          stroke="black"
          stroke-width="4"
        />
        <mask
          fill="white"
          id="path-5-inside-2_3975_16368"
        >
          <rect
            height="6"
            rx="1"
            width="30"
            x="25"
            y="4.9375"
          />
        </mask>
        <rect
          fill="white"
          height="6"
          mask="url(#path-5-inside-2_3975_16368)"
          rx="1"
          stroke="black"
          stroke-width="4"
          width="30"
          x="25"
          y="4.9375"
        />
        <mask
          fill="white"
          id="path-6-inside-3_3975_16368"
        >
          <path
            d="M56.5 6.0625C56.5 5.51022 56.9477 5.0625 57.5 5.0625H59.5C60.6046 5.0625 61.5 5.95793 61.5 7.0625V9.0625C61.5 9.61478 61.0523 10.0625 60.5 10.0625H57.5C56.9477 10.0625 56.5 9.61478 56.5 9.0625V6.0625Z"
          />
        </mask>
        <path
          d="M56.5 6.0625C56.5 5.51022 56.9477 5.0625 57.5 5.0625H59.5C60.6046 5.0625 61.5 5.95793 61.5 7.0625V9.0625C61.5 9.61478 61.0523 10.0625 60.5 10.0625H57.5C56.9477 10.0625 56.5 9.61478 56.5 9.0625V6.0625Z"
          fill="white"
          mask="url(#path-6-inside-3_3975_16368)"
          stroke="black"
          stroke-width="4"
        />
      </svg>
      <div
        class="d-flex"
      >
        <svg
          height="8"
          width="0"
        >
          <rect
            fill="#D4D7DB"
            height="8"
            width="0"
            x="8"
          />
          <rect
            fill="#6D39AC"
            height="8"
            rx="4"
            ry="4"
            width="0"
          />
          <ellipse
            cx="-6"
            cy="4"
            fill="white"
            rx="2"
            ry="2"
          />
        </svg>
        <svg
          height="8"
          width="0"
        >
          <rect
            fill="#D4D7DB"
            height="8"
            width="8"
          />
          <rect
            fill="#D4D7DB"
            height="8"
            rx="4"
            ry="4"
            width="0"
          />
          <ellipse
            cx="-6"
            cy="4"
            fill="white"
            rx="2"
            ry="2"
          />
          <ellipse
            cx="-6"
            cy="4"
            fill="white"
            rx="2"
            ry="2"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;
